tools/monitoring/validate.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import "fmt"
+
+// collectPatterns collects all regexp patterns, including both error message patterns
+// and job name patterns
+func collectPatterns(config *Config) []string {
```

Comment:
this seems like a method on config object?

Reply:
It could be. However, it's only used by the validator and it could be confusing to include it as a method of Config. It's triggered in different workflow than the rest of methods there.

Reply:
yes. But, can be used by others. Rename this to GetAllPatterns()